File: frontend/src/types/safety.ts
export type SafetySummary = {
  total: number;
  safe: number;
  constructive: number;
  toxic: number;
  severe_abuse: number;
  spam: number;
  shielded: number;
};

export type SafetyAnalyzeResult = {
  analyzed_comments: number;
};

export type SafetyComment = {
  id: number;
  text: string;
  username: string | null;
  commented_at: string | null;
};

File: backend/src/veya/api/schemas/safety.py
from datetime import datetime

from pydantic import BaseModel


class SafetyAnalyzeResponse(BaseModel):
    analyzed_comments: int


class SafetySummaryResponse(BaseModel):
    total: int
    safe: int
    constructive: int
    toxic: int
    severe_abuse: int
    spam: int
    shielded: int


class SafetyCommentResponse(BaseModel):
    id: int
    text: str
    username: str | None
    commented_at: datetime | None
